### PR TITLE
scraper: Fix scraper deadlock and make activebeforesb dynamic (not require restart)

### PR DESCRIPTION
**Pull request by jamescowens**
The while loop that controls the quiescent period before scraper becomes active before the SB is due was missing an update on nBeforeSBSleep. This causes the scraper to ignore protocol changes introduced after scraper startup for this parameter. This corrects that and activebeforesb will now be dynamic and not require a scraper restart to change.

There was a scoping issue with the cs_ConvergedScraperStatsCache that was causing the lock to be applied recursively (twice) which leads to a deadlock in a corner case situation when the wallet has a lot of UTXO's and the miner loop time is quite large. (This was discovered by the testnet scraper on an Odroid with a 20k+ UTXO wallet.) With this fix the Odroid has run for a week without a problem.

### CODE REVIEW

File: src/gridcoin/scraper/scraper.cpp [hunk 1]
```diff
@@ -1579,6 +1579,9 @@ void Scraper(bool bSingleShot)
                 _log(logattribute::INFO, "Scraper", "Sleeping for " + ToString(scraper_sleep() / 1000) +" seconds");
 
                 if (!MilliSleep(scraper_sleep())) return;
+
+                sbage = SuperblockAge();
```
Comment (div72) [inline src/gridcoin/scraper/scraper.cpp]: L1577-1578 should be removed in my opinion. Taking the lock on cs_main just for a log message seems a bit wasteful.

Comment (jamescowens) [reply src/gridcoin/scraper/scraper.cpp]: Well... the log message is important. Quite frankly the second call to SuperblockAge() is the really important one. I suppose we can do the first one and then just add scraper_sleep() to it for the second, so we only do it once.

Comment (div72) [reply src/gridcoin/scraper/scraper.cpp]: I like this approach better but couldn't scraper_sleep change?

Comment (jamescowens) [reply src/gridcoin/scraper/scraper.cpp]: The chances of that happening are very small, but might as well make it 100% tight... We also save a lock on cs_ScraperGlobals.

**Review by div72 — APPROVED** [review]
utACK.

**Pull request merged**